docs/src/index.md

```diff
@@ -127,6 +133,60 @@ julia> autodiff(Forward, rosenbrock_inp, BatchDuplicated, BatchDuplicated(x, (dx
 (400.0, (var"1" = -800.0, var"2" = 400.0))
 ```
 
+### Convenience functions
+
+!!! note
+    While the convenience functions discussed below use [`autodiff`](@ref) internally, they are generally more limited in their functionality. Beyond that, these convenience functions may also come with performance penalties; especially if one makes a closure of a multi-argument function instead of calling the appropriate multi-argument [`autodiff`](@ref) function directly.
+
+Key convenience functions for common derivative computations are [`gradient`](@ref) (and its inplace variant [`gradient!`](@ref)) and [`jacobian`](@ref).
+Like [`autodiff`](@ref), the mode (forward or reverse) is determined by the first argument.
+
+The functions [`gradient`](@ref) and [`gradient!`](@ref) compute the gradient of function with vector input and scalar return.
+
+```jldoctest rosenbrock
+julia> gradient(Reverse, rosenbrock_inp, [1.0, 2.0])
+2-element Vector{Float64}:
+ -400.0
+  200.0
+
+julia> # inplace variant
+       dx = [0.0, 0.0];
+       gradient!(Reverse, dx, rosenbrock_inp, [1.0, 2.0])
+2-element Vector{Float64}:
+ -400.0
+  200.0
+
+julia> dx
+2-element Vector{Float64}:
+ -400.0
+  200.0
+
+julia> gradient(Forward, rosenbrock_inp, [1.0, 2.0])
+(-400.0, 200.0)
+
+julia> # in forward mode, we can also optionally pass a chunk size
+       # to specify the number of derivatives computed simulateneously
+       # using vector forward mode
+       gradient(Forward, rosenbrock_inp, [1.0, 2.0], Val(2))
+(-400.0, 200.0)
+```
+
+The function [`jacobian`](@ref) computes the Jacobian of a function vector input and vector return.
+
+```jldoctest rosenbrock
+julia> foo(x) = [rosenbrock_inp(x), prod(x)];
+
+julia> jacobian(Reverse, foo, [1.0, 2.0], Val(2)) # here Val(2) specifies the output size of `foo` since it cannot be statically inferred
+2×2 Matrix{Float64}:
+ -400.0  200.0
+    2.0    1.0
+
+julia> jacobian(Forward, foo, [1.0, 2.0])
+2×2 Matrix{Float64}:
+ -400.0  200.0
+    2.0    1.0
+```
+
```

Comment:
Sorry, one final thing, then let's merge it.

Can you similarly have one more example here of forward/reverse jacobian in vector mode

Reply:
No need to be sorry; I like complete docs :).

Looking at the api docs, there may also be an opportunity to aggregate some of the doc strings of methods with optional arguments into one. They are currently quite scattered which is why I actually didn't know there's Jacobian vector mode for reverse too. 

But, if you don't mind, I'll leave that cleanup for another PR.